pkg/controller/storagecluster/externalstoragecluster_controller.go

```diff
@@ -0,0 +1,265 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	objectreferencesv1 "github.com/openshift/custom-resource-status/objectreferences/v1"
+	ocsv1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1"
+	statusutil "github.com/openshift/ocs-operator/pkg/controller/util"
+	"github.com/operator-framework/operator-sdk/pkg/ready"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/reference"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// ReconcileExternalStorageCluster reconciles the external Storage Cluster
+func (r *ReconcileStorageCluster) ReconcileExternalStorageCluster(sc *ocsv1.StorageCluster) (reconcile.Result, error) {
```

Comment:
Need a check to see if another StorageCluster Object is already present.
If converged StorageCluster is present, it can not be replaced with External StorageCluster and vice-versa.

Reply:
Now we are making changes into the original `Reconcile` method, the said functionality is already present